pkg/client/unversioned/remotecommand/remotecommand.go

```diff
@@ -194,12 +194,19 @@ func (e *Streamer) doStream() error {
 		if err != nil {
 			return err
 		}
-		defer remoteStdin.Reset()
```

Comment:
interesting, the `cp` essentially "intercepts" the EOF, so the remote command never sees it.